src/models/brightcove.js

```diff
@@ -104,6 +111,51 @@ class Brightcove extends Video {
 		};
 	}
 
+	setUpAds() {
+		this.adContainerEl = document.createElement('div');
+		this.containerEl.appendChild(this.adContainerEl);
+		this.adDisplayContainer = new google.ima.AdDisplayContainer(this.adContainerEl, this.el);
+
+		// Create ads loader.
+		this.adsLoader = new google.ima.AdsLoader(this.adDisplayContainer);
+
+		// Sets up bindings for all Ad related handlers
+		this.adsManagerLoadedHandler = this.adsManagerLoadedHandler.bind(this);
+		this.adErrorHandler = this.adErrorHandler.bind(this);
+		this.adEventHandler = this.adEventHandler.bind(this);
+		this.contentPauseRequestHandler = this.contentPauseRequestHandler.bind(this);
+		this.contentResumeRequestHandler = this.contentResumeRequestHandler.bind(this);
+
+		// Listen and respond to ads loaded and error events.
+		this.adsLoader.addEventListener(
+			google.ima.AdsManagerLoadedEvent.Type.ADS_MANAGER_LOADED,
+			this.adsManagerLoadedHandler,
+			false);
+		this.adsLoader.addEventListener(
+			google.ima.AdErrorEvent.Type.AD_ERROR,
+			this.adErrorHandler,
+			false);
+
+		// Request video ads.
+		const adsRequest = new google.ima.AdsRequest();
+		let advertisingUrl = `http://pubads.g.doubleclick.net/gampad/ads?env=vp&gdfp_req=1&impl=s&output=xml_vast2&iu=${this.targeting.site}&sz=${this.targeting.sizes}&unviewed_position_start=1&scp=pos%3D${this.targeting.position}&ttid=${this.targeting.videoId}`;
+		if(this.targeting.brand) {
+			advertisingUrl += `&brand=${encodeURIComponent(this.targeting.brand)}`;
+		}
+
+		adsRequest.adTagUrl = advertisingUrl;
+
+		// Specify the linear and nonlinear slot sizes. This helps the SDK to
+		// select the correct creative if multiple are returned.
+		adsRequest.linearAdSlotWidth = 640;
```

Comment:
Looking into this. Not sure if this should match highest ad dimensions requested: 592x333, or the video size. Currently it is set to 640x360